mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -258,11 +258,7 @@ private[ml] object TreeClassifierParams {
 private[ml] trait DecisionTreeClassifierParams
   extends DecisionTreeParams with TreeClassifierParams
 
-/**
- * Parameters for Decision Tree-based regression algorithms.
- */
-private[ml] trait TreeRegressorParams extends Params {
```

Comment:
You can actually keep this trait and extend HasVarianceImpurity. It would be a no-op like a few other traits here. Although 'redundant', would it avoid any MiMa warnings or very slightly improve compatibility, even if this is private[spark]?

Reply:
actually this trait is still there (just some lines below). The MiMa warnings are there because the returned type of `setImpurity` returns `this.type`, which is now `HasVarianceImpurity` and the same for the setter...

Reply:
Just looking at this one more time ... hm, isn't `this.type` always referring to the current class's type? maybe it's some detail of the generated code and this is essentially a false positive. Or, honestly setImpurity can just be removed in this change too. It's deprecated for all of these classes. I don't mind doing that later too; would just avoid these MiMa exclusions too.

Reply:
yes, it is AFAIK. I think it is a false positive.

I wouldn't remove it in this change because since this is a bug, this PR can be backported to 2.4.1 too. I'd do it in a separate change. We would need anyway to add the MiMa rules for the missing methods, so it doesn't change much on the exclusions either...

Reply:
That's true, but I don't know if we can back-port it because of the binary incompatibility, internal as it may be. I don't know. If it's not an issue then yes it can back port

Reply:
I see. I am not sure how to verify that.